registry/storage/layerreader.go

```diff
@@ -17,15 +18,28 @@ type layerReader struct {
 
 var _ distribution.Layer = &layerReader{}
 
-func (lrs *layerReader) Digest() digest.Digest {
-	return lrs.digest
+func (lr *layerReader) Path() string {
```

Comment:
This should not be exported: the path mapping for a layer is an implementation detail.